### PR TITLE
Emoji convenience methods

### PR DESCRIPTION
**Pull request by PgBiel**
Adds convenience methods (similar to those of RichEmbed) to Emoji, for easier use:

- `setName`
- `addRestrictedRole`
- `addRestrictedRoles`
- `removeRestrictedRole`
- `removeRestrictedRoles`

Still a bit indecisive over either stay using array argument or use rest parameters for the `addRestrictedRoles` method. Let me know if I should use rest parameters.

**Edit:** I can also use both 🤔 
**Edit 2:** If I used both I could also merge the restricted role methods into one method 🤔

### CODE REVIEW

File: src/structures/Emoji.js [hunk 1]
```diff
@@ -124,6 +124,39 @@ class Emoji {
...
+  setName(name) {
+    return this.edit({ name });
+  }
+
+  /**
+   * Add a role to the list of roles that can use this emoji
+   * @param {Role} role The role to add
+   * @returns {Promise<Emoji>}
+   */
+  addRestrictedRole(role) {
```
Comment (GusCaplan) [inline src/structures/Emoji.js]: this should probably just call `this.addRestrictedRoles([role])` for consistancy

**Comment by GusCaplan** [comment]
what about removing restricted roles

File: src/structures/Emoji.js (hunk 1 above)
Comment (PgBiel) [inline src/structures/Emoji.js]: Ok

**Comment by PgBiel** [comment]
I guess I'll do by index

File: src/structures/Emoji.js [hunk 2]
```diff
@@ -157,6 +155,28 @@ class Emoji {
...
+   * Remove multiple roles from the list of roles that can use this emoji
+   * @param {Array<Role>} roles Roles to remove
+   * @returns {Promise<Emoji>}
+   */
+  removeRestrictedRoles(roles) {
+    const newRoles = new Collection(this.roles);
+    for (const role of roles) {
+      if (newRoles.has(role.id)) newRoles.delete(role.id);
+    }
+    return newRoles;
```
Comment (GusCaplan) [inline src/structures/Emoji.js]: return this.edit perhaps

Comment (PgBiel) [inline src/structures/Emoji.js]: I am wordless

**Pull request merged**